index.bs

```diff
@@ -773,17 +753,24 @@ The following steps are taken:
 1. Let _stageValue_ be ToString(_stage_).
 1. RejectIfAbrupt(_stageValue_).
 1. If IsValidStageValue(_stageValue_) is *false*, return a promise rejected with a new *RangeError* exception.
+1. Let _stageEntry_ be GetStage(_entry_, _stageValue_).
+1. If _stageEntry_ is *undefined*, return a promise rejected with a new *TypeError* exception.
+1. Perform UpgradeToStage(_entry_, _stageValue_).
```

Comment:
If you attempt to `resolve()` or `reject()` a entry, it will be bound to the resolution of the provided value (which is promise based), so we automatically discard any previous "stage".